src/Job.php

```diff
@@ -104,7 +104,8 @@ public function setup(): void
 
     public function getSignature(): string
     {
-        return md5(implode('-', [static::class, implode('-', array_keys($this->URLsToProcess))]));
+        return md5(implode('-', [static::class, implode('-',
+            array_keys((array)$this->URLstoProcess))]));
```

Comment:
```suggestion
        if (!$this->URLsToProcess) {
            return md5(static::class);
        }
        return md5(implode('-', [static::class, implode('-', array_keys($this->URLsToProcess))]));
```
If there's nothing to process, then treating it as an array is improper - we should just have a signature that doesn't try to incorporate the URLs in that scenario.